python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -51,6 +51,8 @@
 if TYPE_CHECKING:
     from dagster._core.definitions.asset_graph_subset import AssetGraphSubset
 
+SpecKey = Union[AssetKey, AssetCheckKey]
```

Comment:
@schrockn first use

Reply:
I'm not sure about using "Spec" as an informal supertype for "Asset" and "AssetCheck". "Spec" feels like it's at the same level of generality as "Definition", and that level of generality is much broader than the union of asset and asset check.

You could imagine in the future that we add `JobSpec` as an object that includes all the metadata for a job, without the implementation.

Doesn't roll off the tongue, but the best alternatives I can think of are `AssetKeyOrAssetCheckKey` or `AssetKeyOrCheckKey`.

Reply:
changed to AssetKeyOrCheckKey